heatexchanger/layer.py

```diff
@@ -83,18 +79,20 @@ def setup(self, Nairpipes, Nwaterpipes):
             y_dim >= airpipes.w.sum()
         ]
 
-        for i in range(Nwaterpipes):
-            for j in range(Nairpipes):
-                waterCf.extend([waterpipes.l[i,j] == (j+1)*airpipes.w[0:j+1].prod()**(1./(j+1))])
-                airCf.extend([airpipes.l[j,i] == (i+1)*waterpipes.w[0:i+1].prod()**(1./(i+1))])
+        waterCf = []
+        airCf = []
+        for i in range(Nairpipes):
+            for j in range(Nwaterpipes):
```

Comment:
Note switched i/j indices

Reply:
pretty sure that is not the right indexing... the waterpipe length is the geometric mean of the airpipe lengths over j....

Reply:
It could well be wrong, but try a non-square number of pipes yourself and you'll run into these same errors! Trying to get it working for (3,5) and (5,3) with the fewest changes is what motivated this PR.

Reply:
Well then I'll fix it. Shouldn't take too long.

Reply:
Sounds good. I hope you find these PRs useful.

Reply:
Thanks.